src/services/item/plugins/action/requestExport/itemAction.requestExport.service.ts

```diff
@@ -0,0 +1,379 @@
+  }
+
+  public async request(
+    dbConnection: DBConnection,
+    minimalMember: MinimalMember,
+    itemId: UUID,
+    format: ActionRequestExportFormat,
+  ) {
+    // check member has admin access to the item
+    const member = await this.memberService.get(dbConnection, minimalMember.id);
+    const item = await this.authorizedItemService.getItemById(dbConnection, {
+      actor: minimalMember,
+      itemId,
+      permission: PermissionLevel.Admin,
+    });
+
+    // get last export entry within interval
+    const lastRequestExport = await this.actionRequestExportRepository.getLast(dbConnection, {
+      memberId: member?.id,
+      itemPath: item.path,
+      format,
+    });
+
+    // check if a previous request already created the file and send it back
+    try {
+      if (lastRequestExport) {
+        await this.sendExportLinkInMail(
+          member.toMemberInfo(),
+          item,
+          lastRequestExport.createdAt,
+          format,
+        );
+        return;
+        // the previous exported data does not exist or
+        // is outdated and a new version should be uploaded
+      }
+    } catch (e) {
+      // previous request export exists but not the file
+      // we should create a new export
+      if (!(e instanceof S3FileNotFound)) {
+        throw e;
+      }
+    }
+
+    // create request row
+    const requestExport = await this.actionRequestExportRepository.addOne(dbConnection, {
+      itemPath: item.path,
+      memberId: minimalMember.id,
+      createdAt: new Date().toISOString(),
+      format,
+    });
+
+    // create archive and upload on server
+    await this.createAndUploadArchive(dbConnection, minimalMember, item, requestExport);
+
+    // send email
+    await this.sendExportLinkInMail(member.toMemberInfo(), item, requestExport.createdAt, format);
+
+    return item;
+  }
+
+  /**
+   * Build file path where the archive should be saved on the server
+   */
+  private buildActionFilePath({
+    itemId,
+    datetime,
+    format,
+  }: {
+    itemId: string;
+    datetime: string;
+    format: ActionRequestExportFormat;
+  }): string {
+    return `actions/${itemId}/${format}/${formatDate(datetime, 't')}.zip`;
+  }
+
+  /**
+   * Build name for file within the archive
+   * @param name
+   * @param datetime
+   * @param format
+   * @returns
+   */
+  private buildActionFileName(name: string, datetime: string, format: string): string {
+    return `${name}_${formatDate(datetime, 't')}.${format}`;
+  }
+
+  private async sendExportLinkInMail(
+    member: MemberInfo,
+    item: ItemRaw,
+    archiveDate: string,
+    format: ActionRequestExportFormat,
+  ) {
+    const filepath = this.buildActionFilePath({
+      itemId: item.id,
+      datetime: archiveDate,
+      format,
+    });
+    const link = await this.fileService.getUrl({
+      path: filepath,
+      expiration: EXPORT_FILE_EXPIRATION,
+      downloadName: `${item.name}_${format}_actions_${formatDate(archiveDate, 't')}.zip`,
```

Comment:
The name of the file when stored on s3 is different from the name of the file when downloading. Should they be the same to make debugging and working with them more easy ?